internal/clusterfeature/features/vault/operator.go

```diff
@@ -0,0 +1,403 @@
+		}
+
+		// create Vault manager
+		vaultManager, err := newVaultManager(boundSpec, orgID, clusterID, token)
+		if err != nil {
+			return errors.WrapIf(err, "failed to create Vault manager")
+		}
+
+		// create policy
+		var policy string
+		if boundSpec.CustomVault.Enabled {
+			policy = boundSpec.CustomVault.Policy
+		} else {
+			policy = getDefaultPolicy(orgID)
+		}
+		if err := vaultManager.createPolicy(policy); err != nil {
+			return errors.WrapIf(err, "failed to create policy")
+		}
+		logger.Info("policy created successfully")
+
+		// enable auth method
+		if err := vaultManager.enableAuth(getAuthMethodPath(orgID, clusterID), authMethodType); err != nil {
+			return errors.WrapIf(err, fmt.Sprintf("failed to enabling %s auth method for vault", authMethodType))
+		}
+		logger.Info(fmt.Sprintf("auth method %q enabled for vault", authMethodType))
+
+		// config auth method
+		if _, err := vaultManager.configureAuth(tokenReviewerJWT, kubeConfig.Host, kubeConfig.CAData); err != nil {
+			return errors.WrapIf(err, fmt.Sprintf("failed to configure %s auth method for vault", authMethodType))
+		}
+		logger.Info(fmt.Sprintf("auth method %q enabled for vault", authMethodType))
+
+		// create role
+		if _, err := vaultManager.createRole(boundSpec.Settings.ServiceAccounts, boundSpec.Settings.Namespaces); err != nil {
+			return errors.WrapIf(err, fmt.Sprintf("failed to create role in the auth method %q", authMethodType))
+		}
+		logger.Info(fmt.Sprintf("role created in auth method %q for vault", authMethodType))
+
+	}
+
+	return nil
+}
+
+func getPolicyName(orgID, clusterID uint) string {
+	return fmt.Sprintf("%s_%d_%d", policyNamePrefix, orgID, clusterID)
+}
+
+func (op FeatureOperator) installOrUpdateWebhook(
+	ctx context.Context,
+	logger common.Logger,
+	orgID, clusterID uint,
+	spec vaultFeatureSpec,
+) error {
+	// create chart values
+	pipelineSystemNamespace := viper.GetString(config.PipelineSystemNamespace)
+	var chartValues = &webhookValues{
+		Env: map[string]string{
+			vaultAddressEnvKey: spec.getVaultAddress(),
+			vaultPathEnvKey:    getAuthMethodPath(orgID, clusterID),
+		},
+		NamespaceSelector: namespaceSelector{
+			MatchExpressions: []matchExpressions{
+				{
+					Key:      "name",
+					Operator: "NotIn",
+					Values: []string{
+						kubeSysNamespace,
+						pipelineSystemNamespace,
+					},
+				},
+			},
+		},
+	}
+	valuesBytes, err := json.Marshal(chartValues)
+	if err != nil {
+		logger.Debug("failed to marshal chartValues")
+		return errors.WrapIf(err, "failed to decode chartValues")
+	}
+
+	chartName := getChartName()
+	chartVersion := getChartVersion()
+
+	return op.helmService.ApplyDeployment(
+		ctx,
+		clusterID,
+		pipelineSystemNamespace,
+		chartName,
+		vaultWebhookReleaseName,
+		valuesBytes,
+		chartVersion,
+	)
+}
+
+func getChartName() string {
+	return viper.GetString(config.VaultWebhookChartKey)
+}
+
+func getChartVersion() string {
+	return viper.GetString(config.VaultWebhookChartVersionKey)
+}
+
```

Comment:
feature configuration should not come through viper (it's ok for the time being, however i'd opt for a separate service for gathering configurations coming out of the spec)